core/app/models/spree/order_shipping.rb

```diff
@@ -74,8 +72,11 @@ def ship(inventory_units:, stock_location:, address:, shipping_method:,
       shipment.update_columns(state: 'shipped', shipped_at: Time.current)
     end
 
-    changed(stock_location.fulfillable? && !suppress_mailer) # sort of a hack
-    notify_observers(:shipped, carton)
+    # Wherever sets suppress_mailer should be deciding whether an observer is
+    # added to the order or not?
+    @order.changed(stock_location.fulfillable? && !suppress_mailer)
+    @order.notify_observers(:shipped, carton)
```

Comment:
So I tried something else here, and you can let me know what you think. 

When the carton is created above, I added `@order` as an observer on carton (Carton had become Observable). Then the code here could be `carton.ship(suppress_mailer)`. Carton checks if `stock_location.fulfillable?` and then if it is, notifies all its observers (order). Then when order gets notified it passes the message along to its observers
e.g. inside order

```
def update(change, *args)
  changed(true)
  notify_observers(change, self, *args)
end
```